### PR TITLE
add homework01

### PR DESCRIPTION
**Pull request by jsakhno**
@AlexRozl pl reviewe

### CODE REVIEW

File: homeworks/yaroslav_sakhno/hw01/test.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+class TryTest(Testcase):
+
+    parameters = {
+        'num1': 3,
+        'num2': 0
+    }
+
+    @test
+    def test_add(self,steps, num1, num2):
+        with steps.start('Add'):
```
Comment (AlexRozl) [inline homeworks/yaroslav_sakhno/hw01/test.py]: Unnecessary step, you dont need to check is the function adds correctly

File: homeworks/yaroslav_sakhno/hw01/test.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            assert add(num1,num2) == num1 + num2
+        with steps.start('Result > 0'):
+            try:
+                assert add(num1, num2) > 0
+            except AssertionError:
+                self.skipped('Result < 0)')
+
+    @test
+    def test_divide(self, num1, num2):
```
Comment (AlexRozl) [inline homeworks/yaroslav_sakhno/hw01/test.py]: You have not checked whether the result of division is greater than zero, please add this.

File: homeworks/yaroslav_sakhno/hw01/test.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        try:
+            assert divide(num1, num2) == num1 / num2
+        except ZeroDivisionError:
+            self.passx('passx')
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="standalone parser")
```
Comment (AlexRozl) [inline homeworks/yaroslav_sakhno/hw01/test.py]: Parser description does not have any useful information to understand what are this numbers.

File: homeworks/yaroslav_sakhno/hw01/test.py (hunk 2 above)
Comment (jsakhno) [reply homeworks/yaroslav_sakhno/hw01/test.py]: @AlexRozl fixed

File: homeworks/yaroslav_sakhno/hw01/test.py (hunk 3 above)
Comment (jsakhno) [reply homeworks/yaroslav_sakhno/hw01/test.py]: @AlexRozl fixed

File: homeworks/yaroslav_sakhno/hw01/test.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    @test
+    def test_divide(self, steps, num1, num2):
+        with steps.start('Add'):
+            try:
+                assert divide(num1, num2) == num1 / num2
+            except ZeroDivisionError:
+                self.passx('passx')
+        with steps.start('Result > 0'):
+            try:
+                assert add(num1, num2) > 0
```
Comment (AlexRozl) [inline homeworks/yaroslav_sakhno/hw01/test.py]: Why you test "add" function in "test_divide"?

File: homeworks/yaroslav_sakhno/hw01/test.py (hunk 1 above)
Comment (jsakhno) [reply homeworks/yaroslav_sakhno/hw01/test.py]: @AlexRozl sorry, but this step is necessary for me.

File: homeworks/yaroslav_sakhno/hw01/test.py (hunk 4 above)
Comment (jsakhno) [reply homeworks/yaroslav_sakhno/hw01/test.py]: @AlexRozl fixed

Comment (jsakhno) [reply homeworks/yaroslav_sakhno/hw01/test.py]: @AlexRozl one more)

File: homeworks/yaroslav_sakhno/hw01/test.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import sys
+import argparse
+from pyats.aetest import Testcase, test, main
+from calculation import add, divide
+
+
+class TryTest(Testcase):
+
+    parameters = {
```
Comment (extsoft) [inline homeworks/yaroslav_sakhno/hw01/test.py]: @jsakhno Please set default arguments in the args parser and remove this dictionary.

File: homeworks/nazar_kostiv/pyats01_hw.py [hunk 6]
```diff
@@ -52,4 +52,4 @@ def test_divide(self, steps, num1, num2):
     parser.add_argument('-num1', dest='num1', type=int, required=False, default=parameters['num1'])
     parser.add_argument('-num2', dest='num2', type=int, required=False, default=parameters['num2'])
     args, sys.argv[1:] = parser.parse_known_args(sys.argv[1:])
-    main(num1=args.num1, num2=args.num2)
+    main(num1=args.num1, num2=args.num2)
```
Comment (extsoft) [inline homeworks/nazar_kostiv/pyats01_hw.py]: @jsakhno an empty line is required at the end of the file.

File: homeworks/yaroslav_sakhno/hw01/test.py (hunk 5 above)
Comment (jsakhno) [reply homeworks/yaroslav_sakhno/hw01/test.py]: @extsoft fixed

File: homeworks/nazar_kostiv/pyats01_hw.py (hunk 6 above)
Comment (jsakhno) [reply homeworks/nazar_kostiv/pyats01_hw.py]: @extsoft fixed

**Comment by extsoft** [comment]
@jsakhno please rebase master branch into this one.

**Pull request merged**